Source/WebCore/html/OffscreenCanvas.cpp

```diff
@@ -180,8 +180,12 @@ void OffscreenCanvas::setHeight(unsigned newHeight)
 
 void OffscreenCanvas::setSize(const IntSize& newSize)
 {
+    IntSize oldSize = size();
     CanvasBase::setSize(newSize);
     reset();
+
+    if (m_context && m_context->isGPUBased() && oldSize != size())
+        downcast<GPUBasedCanvasRenderingContext>(*m_context).reshape(width(), height());
```

Comment:
`reshape` checks for `width` and `height` being different to it's current value, so you could skip the `oldSize != size()`